packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -90,15 +102,22 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
     final eventStoragePath = await _pathProvider?.getApplicationSupportPath();
     final eventStore = DartQueuedItemStore(eventStoragePath);
 
-    await _analyticsClient.init(
+    final analyticsClient = AnalyticsClient(
+      storageScope: AmplifyStorageScope.awsPinpointAnalyticsPlugin,
+      deviceContextInfoProvider: _deviceContextInfoProvider,
+      legacyNativeDataProvider: _legacyNativeDataProvider,
+      secureStorageFactory: _secureStorageFactory,
+    );
```

Comment:
`AnalyticsClient` will be used outside of Analytics in Push Notifications to interact with Pinpoint. 

While Push Notifications is internal so we have more control. I do prefer for external package usage to be strongly typed and controlled. 

Switching to taking in a storage instance directly could suggest or allow for storage instances that do not have the proper scope to be passed in.

Reply:
If the goal is to prevent using AmplifySecureStorage with an improper scope, there are better ways to do that.

First, we should mark the `AmplifySecureStorage()` constructor as internal. This will discourage use of creating an instance with a scope other than AmplifyStorageScope because the only way to do so would require using an internal member. `AmplifySecureStorage.factoryFrom` only accepts `AmplifyStorageScope`.

```dart
final s = AmplifySecureStorage.factoryFrom()(
  AmplifyStorageScope.awsCognitoAuthPlugin, // a string would result in a compile time error
);
```

If we feel that this is not enough, we can update `AmplifySecureStorageConfig()` to accept type `AmplifyStorageScope`. 

`AnalyticsClient` is not the correct place to try to enforce this.

Reply:
Sounds good thanks.  Updated to accept SecureStorageInterface in AnalyticsClient.